File: client/src/app/models/timeline.model.ts
export enum ProtectLevel {
  public,
  protect
}

export enum PeriodGroupLevel {
  any,
  byDay,
  byMonth,
  byYear
}

export interface Timeline {
  username: string;
  title: string;
  topicKey: string;
  protectLevel: ProtectLevel;
  accessKey?: string;
  periodGroupLevel: PeriodGroupLevel;
}

File: server/TimelinesAPI/Models/TimelineModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using TimelinesAPI.DataVaults;

namespace TimelinesAPI.Models
{
    public class TimelineModel
    {
		public string Username { get; set; }
		public string TopicKey { get; set; }
		public string Title { get; set; }
		[JsonConverter(typeof(StringEnumConverter))]
		public ProtectLevel ProtectLevel { get; set; }
		public string AccessKey { get; set; }
	    [JsonConverter(typeof(StringEnumConverter))]
		public PeriodGroupLevel PeriodGroupLevel { get; set; }
		public bool IsCompleted { get; set; }
        public DateTime StartTime { get; set; }
    }
}
